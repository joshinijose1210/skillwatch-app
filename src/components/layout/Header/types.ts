export interface NotificationData {
    toastMessage: string;
    onClick: (e: any) => void;
}

export interface LocationData {
    action: string;
    from: string;
}

export interface BreadcrumbItemData {
    match: any;
    location: any;
    key: string;
    breadcrumb: any;
}

export interface MyComponentProps {
    route: string;
    employeeName: string;
    breadCrumbValue: string;
    breadcrumbLength: boolean;
}

export interface ReviewCycle {
    organisationId: number;
    reviewCycleId: number;
    startDate: number;
    endDate: number;
    publish: boolean;
    lastModified: number;
    selfReviewStartDate: number;
    selfReviewEndDate: number;
    managerReviewStartDate: number;
    managerReviewEndDate: number;
    checkInWithManagerStartDate: number;
    checkInWithManagerEndDate: number;
}
