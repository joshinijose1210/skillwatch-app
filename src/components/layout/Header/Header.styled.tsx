import { Breadcrumb as MedlyBreadcrumb, Text, Toast } from '@medly-components/core';
import { SvgIcon } from '@medly-components/icons';
import { defaultTheme } from '@theme';
import styled from 'styled-components';

export const Header = styled('header')`
    min-height: 6.2rem;
    border-bottom: 1px solid ${({ theme }) => theme.customColors.borderColor};
    background-color: ${defaultTheme.colors.white};
    box-shadow: 0 0.2rem 8 ${defaultTheme.colors.grey[50]};
    display: flex;
    align-items: center;
    justify-content: space-between;
    position: relative;

    @media (max-width: 1279px) {
        padding-left: 1.6rem;
        padding-right: 1.6rem;
    }

    @media (min-width: 1280px) {
        padding-left: calc(1.6rem + (((100vw - 1280px) / 160) * 16));
        padding-right: calc(2.8rem + (((100vw - 1280px) / 160) * 16));
    }

    @media (min-width: 1440px) {
        padding-left: 2.4rem;
        padding-right: 2.4rem;
    }
`;

export const LeftSide = styled('div')`
    display: flex;
    align-items: center;

    & > * {
        margin-right: ${({ theme }) => theme.spacing.S2};
    }

    svg {
        font-size: 2rem;
        * {
            fill: #999;
        }
    }
`;

export const DashboardHeading = styled.div`
    display: flex;
    justify-content: space-between;
`;

export const Welcome = styled(Text)`
    font-size: 16px;
    color: ${({ theme }) => theme.pageTitleColor};
    line-height: 35px;
    position: relative;
    left: -0.8rem;
`;

export const RightSide = styled('div')`
    display: flex;
    align-items: center;
    & > * {
        margin-left: ${({ theme }) => theme.spacing.S2};
    }
`;

export const BreadcrumbItem = styled(MedlyBreadcrumb.Item)`
    margin-left: -0.6rem;
    ${Text.Style} {
        color: #999;
        font-size: ${defaultTheme.font.variants.h6.fontSize};
        font-weight: ${defaultTheme.font.weights.Medium};
        line-height: 2.4rem;
        letter-spacing: -0.02rem;
    }
`;

export const StyledToast = styled(Toast)`
    cursor: pointer;
    margin-right: 1.5rem;
    transition: transform 0.2s;
    box-shadow: 1px 0 6px ${defaultTheme.colors.grey[400]};
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    border-left: 0.4rem solid orange;

    div:first-child {
        align-items: center;
    }

    div:nth-child(1) {
        background-color: rgba(255, 165, 0, 0.1);
    }

    div:nth-child(2) {
        padding: 0.8rem 1.6rem;
    }

    div:nth-child(3) {
        display: none;
    }

    svg:nth-child(1) {
        * {
            fill: orange;
        }
    }

    svg[title='toast-close-icon'] {
        margin-left: 1rem;
    }
`;

export const ToastContent = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: transparent !important;
    ${SvgIcon} {
        transition: all 100ms ease-out;
        background-color: transparent;
        * {
            fill: ${({ theme }) => theme.colors.black};
        }
        &:hover {
            background-color: ${({ theme }) => theme.colors.grey[100]};
        }
    }
`;
