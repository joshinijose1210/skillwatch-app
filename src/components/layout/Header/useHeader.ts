import { routeConstants } from '@constants';
import { Option } from '@medly-components/core/dist/es/components/MultiSelect/types';
import { useAppSelector } from '@slice';
import { updateReviewCycleFilter } from '@slice/reviewCycleFilter';
import { dashboardFilter } from '@slice/dashboardFilter';
import { analyticsFilter } from '@slice/analyticsFilter';
import { useGetDesignationsQuery, useGetReviewCycleDataMutation, useGetTimelineMutation } from '@slice/services';
import { format } from 'date-fns';
import { useCallback, useEffect, useState } from 'react';
import { useDispatch } from 'react-redux';
import { useLocation, useNavigate, useParams } from 'react-router-dom';
import { ReviewCycle } from './types';
import { reviewStatusConstants } from '@constants/reviewStatusConstants';

export const useHeader = () => {
    const dispatch = useDispatch();
    const { ActivePage } = useParams();
    const { reviewCycleList } = useAppSelector(state => state.reviewCycleFilter);
    const analyticsReviewCycle = useAppSelector(state => state.analyticsReviewFilter.reviewCycle);
    const dashboardReviewCycle = useAppSelector(state => state.dashboardReviewFilter.reviewCycle);
    const path = useLocation().pathname;
    const [toastMsg, setToastMsg] = useState('');
    const [toastVisible, setToastVisible] = useState(false);

    const [showCycleFilter, setShowCycleFilter] = useState(true);
    const [isSingleReviewCycle, setIsSingleReviewCycle] = useState(false);
    const [selectedReviewCycle, setSelectedReviewCycle] = useState(0);
    const userDetails = useAppSelector(state => state.user);
    const navigate = useNavigate();
    const [filterOptions, setFilterOptions] = useState<Option[]>([]);
    const [getTimeline, { data }] = useGetTimelineMutation();
    const [getReviewCycleData, { data: reviewCyclesData, isSuccess: reviewCyclesLoaded }] = useGetReviewCycleDataMutation();
    const handleToast = (e: any) => {
        if (e.target.id === 'child-element') {
            setToastMsg('');
        } else if (path === '/') {
            toastMsg === reviewStatusConstants.SelfReviewPending
                ? navigate(routeConstants.selfReview)
                : toastMsg === 'Manager review pending'
                ? navigate(routeConstants.managerReview)
                : toastMsg === reviewStatusConstants.CheckInWithManagerPending
                ? navigate(`${routeConstants.myCheckInWithManager}/${1}`)
                : navigate(`${routeConstants.checkInWithTeamMember}/${1}`);
        }
    };

    const { data: designationsData } = useGetDesignationsQuery(
        {
            path: '',
            params: [
                { name: 'organisationId', value: userDetails.organisationId },
                { name: 'page', value: 99 },
                { name: 'limit', value: 10 },
                { name: 'searchText', value: '' }
            ]
        },
        { refetchOnMountOrArgChange: true, skip: path !== '/designations' }
    );

    const handleDropdownChange = useCallback(
        (value: any) => {
            const updatedPath = path.replace(/\/\d+$/, '/1');
            let selectedValues = value;
            if (path === '/' || path === '/reports/analytics') {
                selectedValues = [value];
                setSelectedReviewCycle(value);
                if (path === '/') {
                    dispatch(dashboardFilter(String(value)));
                } else {
                    dispatch(analyticsFilter(String(value)));
                }
            } else {
                if (selectedValues.length === 0) {
                    dispatch(updateReviewCycleFilter({ reviewCycleList: [-99] }));
                } else {
                    dispatch(updateReviewCycleFilter({ reviewCycleList: selectedValues }));
                }
            }
            navigate(updatedPath);
        },
        [dispatch, navigate, path]
    );

    useEffect(() => {
        if (reviewCyclesData && reviewCyclesLoaded) {
            let activeReviewCycle = -99;
            const list: Option[] =
                reviewCyclesData?.reviewCycles?.map((cycle: ReviewCycle) => {
                    if (cycle.publish) {
                        activeReviewCycle = cycle.reviewCycleId;
                    }
                    return {
                        value: cycle.reviewCycleId,
                        label: `${format(new Date(cycle.startDate), 'do MMM yyyy')} - ${format(new Date(cycle.endDate), 'do MMM yyyy')}`
                    };
                }) ?? [];
            setFilterOptions(list);
            if (path === '/') {
                setSelectedReviewCycle(Number(dashboardReviewCycle));
                if (!dashboardReviewCycle) {
                    dispatch(dashboardFilter(String(activeReviewCycle)));
                }
            }
            if (path === '/reports/analytics') {
                setSelectedReviewCycle(Number(analyticsReviewCycle));
                if (!analyticsReviewCycle) {
                    dispatch(analyticsFilter(String(activeReviewCycle)));
                }
            }
            if (reviewCycleList.length === 0) {
                dispatch(updateReviewCycleFilter({ reviewCycleList: [activeReviewCycle] }));
            }
        }
    }, [dispatch, reviewCyclesData, reviewCycleList, reviewCyclesLoaded]);

    useEffect(() => {
        if (path === '/' && data?.[0] && Object.keys(data[0]).length) {
            const [{ empDetails }] = data;
            if (
                new Date(data[0].selfReviewStartDate) < new Date() &&
                new Date(data[0].selfReviewEndDate + 86400000) > new Date() &&
                !data[0].selfReviewPublish
            ) {
                setToastMsg('Self Review pending');
            } else if (
                new Date(data[0].managerReviewStartDate) < new Date() &&
                new Date(data[0].managerReviewEndDate + 86400000) > new Date() &&
                data[0].empDetails &&
                data[0].empDetails.length > 0
            ) {
                let showMsg = false;
                for (let i = 0; i < empDetails.length; i++) {
                    let publish = false;
                    if (Number(userDetails.id) === empDetails[i].firstManagerId && !empDetails[i].firstManagerReviewPublish) {
                        publish = empDetails[i].firstManagerReviewPublish;
                    } else if (Number(userDetails.id) === empDetails[i].secondManagerId && !empDetails[i].secondManagerReviewPublish) {
                        publish = empDetails[i].secondManagerReviewPublish;
                    } else {
                        publish = true;
                    }
                    if (!publish) {
                        showMsg = true;
                        break;
                    }
                }

                showMsg && setToastMsg('Manager review pending');
            } else if (
                new Date(data[0].checkInWithManagerStartDate) < new Date() &&
                new Date(data[0].checkInWithManagerEndDate + 86400000) > new Date()
            ) {
                let msg = 'Check-in with Manager pending',
                    flag = false;
                if (data[0].empDetails && data[0].empDetails.length > 0) {
                    for (let i = 0; i < empDetails.length; i++) {
                        if (!empDetails[i].checkInWithManagerPublish) {
                            msg = 'Check-in with Team Members is pending';
                            flag = true;
                            break;
                        }
                    }
                }
                if (!flag && data[0].checkInWithManagerPublish) {
                    msg = '';
                }
                setToastMsg(msg);
            }
        } else if (path === '/designations' && designationsData && designationsData.unlinkedDesignationsCount !== 0) {
            setToastMsg('Please add teams to the existing designations');
        } else {
            setToastMsg('');
        }
    }, [userDetails.id, data, path, designationsData]);

    useEffect(() => {
        if (path === '/') {
            getTimeline({
                path: '',
                params: [
                    { name: 'organisationId', value: userDetails.organisationId },
                    { name: 'reviewToId', value: userDetails.id }
                ]
            });
        }
    }, [path, getTimeline, userDetails.organisationId, userDetails.id]);

    useEffect(() => {
        if (toastMsg.length !== 0) {
            setToastVisible(true);
        }
    }, [path, toastMsg.length]);

    useEffect(() => {
        const clickOutSideDropdown = document.getElementById('click-outside-dropdown');
        setIsSingleReviewCycle(false);
        if (
            path === '/' ||
            path === '/reports/analytics' ||
            path === `${routeConstants.threeSixtyDegreeFeedback}/submitted-feedback/${ActivePage}` ||
            path === `${routeConstants.threeSixtyDegreeFeedback}/received-feedback/${ActivePage}` ||
            path === `/request-Feedback/${ActivePage}` ||
            path === `/reports/feedback/${ActivePage}` ||
            path === `/reports/performance-review/${ActivePage}` ||
            path === `/performance-review/self-review` ||
            path === `/performance-review/team's-review` ||
            path === '/performance-review/manager-review' ||
            path === `/performance-review/manager-check-in/${ActivePage}` ||
            path === `/performance-review/team's-check-in/${ActivePage}` ||
            path === `/suggestions/${ActivePage}` ||
            path === `${routeConstants.threeSixtyDegreeFeedback}/${ActivePage}`
        ) {
            getReviewCycleData({
                path: 'getAll',
                params: [{ name: 'organisationId', value: userDetails.organisationId }]
            });
            clickOutSideDropdown?.click();
            if (path === '/' || path === '/reports/analytics') {
                setIsSingleReviewCycle(true);
            } else {
                setShowCycleFilter(true);
            }
        } else {
            setShowCycleFilter(false);
        }
    }, [ActivePage, getReviewCycleData, path, reviewCycleList, userDetails.organisationId]);

    return {
        path,
        toastMsg,
        data,
        handleToast,
        toastVisible,
        reviewCycleList,
        handleDropdownChange,
        filterOptions,
        showCycleFilter,
        isSingleReviewCycle,
        selectedReviewCycle
    };
};
