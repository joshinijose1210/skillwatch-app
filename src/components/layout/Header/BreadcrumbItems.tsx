import { routeConstants } from '@constants';
import { Breadcrumb as MedlyBreadcrumb, Text } from '@medly-components/core';
import { sliceAndCapitalize } from '@utils/sliceAndCapitalize';
import { useCallback } from 'react';
import { useLocation, useNavigate, useParams } from 'react-router-dom';
import { BreadcrumbItem } from './Header.styled';

import { LocationData, MyComponentProps } from './types';
import { useHeader } from './useHeader';
import { FeedbackPageTabValues } from '@pages/MyFeedback/constants';

export const BreadcrumbItems: React.FC<MyComponentProps> = ({ route, employeeName, breadCrumbValue, breadcrumbLength }) => {
    const { path } = useHeader();
    const { ActivePage, id } = useParams();
    const state = useLocation().state as LocationData;
    const navigate = useNavigate();

    const handleClick = useCallback(() => {
        if ([`${routeConstants.checkInWithTeamMember}`].includes(route)) {
            navigate(`${routeConstants.checkInWithTeamMember}/${ActivePage || 1}`);
            return;
        }
        if (path.includes('/reports/analytics')) {
            navigate(routeConstants.analytics);
            return;
        }
        // request feedback page does not exist anymore, so re-routing back to 360 degree feedback page
        if (path.includes(`${routeConstants.requestFeedback}/1`)) {
            navigate(`${routeConstants.threeSixtyDegreeFeedback}/1`);
            return;
        }
        if (
            path.includes('add-goals') ||
            path.includes(routeConstants.selfReview) ||
            path.includes(routeConstants.myManagerReview) ||
            path.includes('/edit-review-cycle') ||
            path.includes('/view-review-cycle') ||
            path.includes('/add-review-cycle') ||
            path.includes('/review-summary')
        ) {
            navigate(-1);
            return;
        }
        if ([`${routeConstants.reviewTimeline}`].includes(route)) {
            navigate(routeConstants.reviewTimeline);
            return;
        }

        if (path.includes(routeConstants.managerReview) || path.includes('view-team-feedback')) {
            navigate(-1);
            return;
        }
        if (!['/reports', '/configuration', '/performance-review', routeConstants.helpAndTraining].includes(route)) {
            navigate(`${route}/${ActivePage}`);
        }
        if (path.includes('/roles-&-permissions') && state && state.from === 'onboardingFlow') {
            navigate(routeConstants.firstRoleRedirect);
        }
        if (path.includes('/dashboard')) {
            navigate(routeConstants.login);
        }
        if (route.includes('/edit-employee')) {
            navigate(`${routeConstants.employeeManagement}/${ActivePage}/edit-employee/${id}`);
        }
    }, [ActivePage, id, navigate, path, route, state]);

    // for handling when current page is /360-degree-feedback and myTab is not a valid /360-degree-feedback page tab id
    if (path.includes(routeConstants.threeSixtyDegreeFeedback)) {
        const myTab = sessionStorage.getItem('myTab');

        if (myTab && !FeedbackPageTabValues.includes(myTab)) {
            sessionStorage.setItem('myTab', 'submitted');
        }
    }

    if (!path.includes('/edit-role-&-permissions')) {
        sessionStorage.removeItem('modulePermissions');
        sessionStorage.removeItem('tempCheckboxState');
    }

    return (
        <MedlyBreadcrumb key={route}>
            <BreadcrumbItem
                key={route}
                onClick={handleClick}
                disabled={
                    breadcrumbLength
                        ? `/app${route}` == window.location.pathname ||
                          ['/employees/edit-employee'].includes(route) ||
                          window.location.pathname.includes(`/${ActivePage}`)
                        : `/app${route}` == window.location.pathname ||
                          ['/reports', '/configuration', '/performance-review', routeConstants.helpAndTraining].includes(route)
                }
            >
                {/* If the route string does not contain any forward slashes or contains more than one forward slash, the expression will return false */}
                {(breadcrumbLength ? route == '/' || route.match(/\//g)?.length === 1 : route == '/') ? (
                    <Text textVariant="h4">{''}</Text>
                ) : (
                    <Text textVariant="h4">{sliceAndCapitalize(employeeName) || sliceAndCapitalize(breadCrumbValue)}</Text>
                )}
            </BreadcrumbItem>
        </MedlyBreadcrumb>
    );
};
