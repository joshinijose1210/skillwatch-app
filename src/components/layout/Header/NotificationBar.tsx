import { Text } from '@medly-components/core';
import { ClearIcon } from '@medly-components/icons';
import { StyledToast, ToastContent } from './Header.styled';
import { NotificationData } from './types';

export const NotificationBar = ({ toastMessage, onClick }: NotificationData) => {
    return (
        <StyledToast
            id={1}
            variant="info"
            hideCloseIcon={true}
            onClick={onClick}
            message={
                <ToastContent>
                    <Text textVariant="body2" textWeight="Medium">
                        {toastMessage}
                    </Text>
                    <ClearIcon id="child-element" title="toast-close-icon" size="XS" variant="solid" />
                </ToastContent>
            }
        />
    );
};
