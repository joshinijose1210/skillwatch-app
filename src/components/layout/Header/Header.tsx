import { StyledMultiSelect, StyledSingleSelect } from '@common';
import { AvatarMenu } from '@components/reusableComponents/AvatarMenu';
import { WithStyle } from '@medly-components/utils';
import { useAppSelector } from '@slice';

import { FC, useState } from 'react';
import useBreadcrumbs from 'use-react-router-breadcrumbs';
// this is not using an alias import for now as a temporary fix for running test cases
import { ReactComponent as ArrowRight } from '../../../constants/images/icons/arrow-right.svg';
import { BreadcrumbItems } from './BreadcrumbItems';
import * as Styled from './Header.styled';
import { NotificationBar } from './NotificationBar';
import { BreadcrumbItemData } from './types';
import { useHeader } from './useHeader';
import TooltipDropdown from '@components/reusableComponents/TooltipDropdown';
import { OptionsType } from '@components/reusableComponents/TooltipDropdown/types';

export const Header: FC & WithStyle = () => {
    const {
        toastMsg,
        handleToast,
        toastVisible,
        path,
        handleDropdownChange,
        reviewCycleList,
        selectedReviewCycle,
        filterOptions,
        showCycleFilter,
        isSingleReviewCycle
    } = useHeader();
    const [open, setOpen] = useState(false);
    const breadcrumbs = useBreadcrumbs();
    const userDetails = useAppSelector(state => state.user);

    const keyToRemovePattern = /^\/employees\/(\d+)\/edit-employee\/(\d+)$|\/\d{1,2}$/;
    const updatedBreadcrumbs = breadcrumbs.filter(item => !keyToRemovePattern.test(item.key));

    return (
        <Styled.Header id="click-outside-dropdown">
            <Styled.LeftSide>
                {userDetails.phoneNumber && userDetails.employeeId && (
                    <>
                        {updatedBreadcrumbs.map((routeObj: BreadcrumbItemData, index: number) => {
                            const breadCrumb = routeObj.breadcrumb && (routeObj.breadcrumb as any).props.children;
                            const userName = breadCrumb.includes('%20')
                                ? `${breadCrumb.split(' ')[0].replace(/%20/g, ' ')}-${breadCrumb.split(' ')[1]}`
                                : breadCrumb.includes('Check')
                                ? `${breadCrumb}`.replace(/(C|c)heck (i|I)n/g, 'Check-in')
                                : breadCrumb.includes('check')
                                ? `${breadCrumb}`.replace(/(C|c)heck (i|I)n/g, 'Check-in')
                                : breadCrumb.includes('(')
                                ? `${breadCrumb}`.replace(/(\(\w+\s\w+)\s/g, '$1-')
                                : breadCrumb.includes('360 degree feedback')
                                ? '360-Degree Feedback'
                                : '';

                            const empName = userName.replace(/%20/g, ' ');
                            return index == 0 || index == updatedBreadcrumbs.length - 1 ? (
                                <>
                                    <BreadcrumbItems
                                        breadCrumbValue={breadCrumb}
                                        route={routeObj.key}
                                        employeeName={empName}
                                        breadcrumbLength={true}
                                    />
                                </>
                            ) : (
                                <>
                                    <BreadcrumbItems
                                        breadCrumbValue={breadCrumb}
                                        route={routeObj.key}
                                        employeeName={empName}
                                        breadcrumbLength={false}
                                    />

                                    <ArrowRight />
                                </>
                            );
                        })}
                    </>
                )}
                {path === '/' && (
                    <Styled.Welcome textVariant="h3" textWeight="Medium">
                        Welcome, {userDetails.firstName.charAt(0).toUpperCase() + userDetails.firstName.slice(1)}
                    </Styled.Welcome>
                )}
            </Styled.LeftSide>
            {toastVisible && toastMsg.length !== 0 && <NotificationBar toastMessage={toastMsg} onClick={handleToast} />}

            <Styled.RightSide>
                {!userDetails.isSuperAdmin && isSingleReviewCycle ? (
                    <>
                        <StyledSingleSelect
                            variant="outlined"
                            options={filterOptions}
                            placeholder="Select Review Cycle"
                            label="Review cycle"
                            value={selectedReviewCycle}
                            onChange={handleDropdownChange}
                            size="M"
                            minWidth="28rem"
                        />
                        <TooltipDropdown
                            dataIds={['review cycle-input']}
                            values={
                                selectedReviewCycle && filterOptions.length
                                    ? filterOptions.filter((item: OptionsType) => selectedReviewCycle === item.value)
                                    : []
                            }
                        />
                    </>
                ) : (
                    !userDetails.isSuperAdmin &&
                    showCycleFilter && (
                        <>
                            <StyledMultiSelect
                                id="dropdown-menu"
                                options={filterOptions}
                                variant="outlined"
                                placeholder="Select Review Cycle"
                                label="Review Cycle"
                                onChange={val => val && handleDropdownChange(val)}
                                values={reviewCycleList}
                                size="M"
                                minWidth="28rem"
                            />
                            <TooltipDropdown
                                dataIds={['dropdown-menu-input', 'dropdown-menu-count-chip']}
                                values={
                                    reviewCycleList?.length && filterOptions.length
                                        ? filterOptions.filter(
                                              (item: OptionsType) =>
                                                  typeof item.value === 'number' && (reviewCycleList as number[]).includes(item.value)
                                          )
                                        : []
                                }
                            />
                        </>
                    )
                )}

                <AvatarMenu open={open} setOpen={setOpen} />
            </Styled.RightSide>
        </Styled.Header>
    );
};

Header.displayName = 'Header';
Header.Style = Styled.Header;

export default Header;
