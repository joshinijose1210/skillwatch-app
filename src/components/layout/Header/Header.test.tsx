import { fireEvent, RTKRender, screen } from '@utils/test-utils';
import Header from './Header';
import { routeConstants } from '@constants';

const renderer = (pathname: string) => {
    return RTKRender(<Header />, { initialEntries: [{ pathname }] });
};

const mockedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedNavigate
}));

describe('Header', () => {
    it('should render properly', () => {
        const { container } = RTKRender(<Header />);
        expect(container).toMatchSnapshot();
    });

    it('should render properly for home path', () => {
        jest.mock('use-react-router-breadcrumbs', () => {
            return jest.fn(() => {
                jest.fn(() => []);
            });
        });
        renderer('/');
        expect(screen.getByText(/Welcome/i)).toBeInTheDocument();
    });

    it('should render properly for received path', () => {
        jest.mock('use-react-router-breadcrumbs', () => {
            return jest.fn(() => {
                jest.fn(() => []);
            });
        });
        renderer(`${routeConstants.threeSixtyDegreeFeedback}/undefined/view-feedback`);
        expect(screen.getByText('My Feedback')).toBeInTheDocument();
    });

    it('should navigate onClick', () => {
        jest.mock('use-react-router-breadcrumbs', () => {
            return jest.fn(() => {
                jest.fn(() => []);
            });
        });
        renderer(`${routeConstants.threeSixtyDegreeFeedback}/1/view-feedback`);
        expect(screen.getByText('My Feedback')).toBeInTheDocument();
        fireEvent.click(screen.getByText('My Feedback'));
        expect(mockedNavigate).toHaveBeenCalledWith(`${routeConstants.threeSixtyDegreeFeedback}/undefined`);
    });
});
