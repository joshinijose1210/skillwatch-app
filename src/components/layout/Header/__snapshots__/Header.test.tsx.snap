// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<div>
  <header
    class="sc-16ik3hp-0 gjqjjH"
    id="click-outside-dropdown"
  >
    <div
      class="sc-16ik3hp-1 hETOWa"
    >
      <ol
        class="sc-1frtvp3-0 iXOpqi"
      >
        <li
          class="sc-qumdxm-0 jWCLpb sc-16ik3hp-5 hXnkuL"
          textcolor="[object Object]"
        >
          <h4
            class="sc-1ntgl3l-0 iEGCWU"
            textalign="initial"
            textcolor="inherit"
            textvariant="h4"
          />
        </li>
      </ol>
      <h3
        class="sc-1ntgl3l-0 kcLBOi sc-16ik3hp-3 jgCRvi"
        textalign="initial"
        textcolor="inherit"
        textvariant="h3"
        textweight="Medium"
      >
        Welcome, 
        ABC
      </h3>
    </div>
    <div
      class="sc-16ik3hp-4 geYifm"
    >
      <div
        class="sc-1hcsztp-0 hyshfb sc-tkiih6-13 cecHDR"
        id="review cycle-wrapper"
        minwidth="28rem"
        variant="outlined"
      >
        <div
          class="sc-17d4s39-0 fpSDTg"
          id="review cycle-input-wrapper"
          minwidth="28rem"
        >
          <div
            class="sc-19px6zp-0 BrSGZ"
            filled="[object Object]"
            fusion="[object Object]"
            minrows="1"
            outlined="[object Object]"
            textvariant="[object Object]"
            variant="outlined"
          >
            <div
              class="sc-lhr232-0 btNOKw"
              variant="outlined"
            >
              <input
                aria-describedby="review cycle-helper-text"
                autocomplete="off"
                class="sc-hb99wz-0 jaEbhp"
                data-tooltip-id="review cycle-input"
                errortext=""
                id="review cycle-input"
                inputmode="none"
                inputsize="M"
                placeholder="Select Review Cycle"
                type="text"
                value=""
                variant="outlined"
              />
              <label
                class="sc-1du51u0-0 eMZrST"
                for="review cycle-input"
                inputwidth="0"
                variant="outlined"
              >
                Review cycle
              </label>
            </div>
            <svg
              class="sc-1pf3x8a-0 klDMsE sc-dynlvq-0 bdmviR"
              fill="none"
              height="1em"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.792 14.97a1 1 0 0 1-1.584 0l-3.165-4.11a1 1 0 0 1 .793-1.61h6.328a1 1 0 0 1 .793 1.61l-3.165 4.11z"
                fill="#000"
              />
            </svg>
          </div>
          
        </div>
      </div>
      
      <div
        class="sc-1298oxm-0 dQDoHN"
      >
        <div
          class="sc-1298oxm-3 kwZxBE"
        >
          <div
            class="sc-1dzehji-0 hjLOhC"
            defaults="[object Object]"
          >
            <strong
              class="sc-1ntgl3l-0 fzLUlA"
              textalign="initial"
              textcolor="inherit"
              textweight="Strong"
            >
              A
            </strong>
            <strong
              class="sc-1ntgl3l-0 fzLUlA"
              textalign="initial"
              textcolor="inherit"
              textweight="Strong"
            >
              D
            </strong>
          </div>
        </div>
        <div
          class="sc-1298oxm-2 ekJHDu inactive"
        >
          <ul
            class="sc-1298oxm-1 jVMYcc"
          >
            <li
              class="sc-1298oxm-4 giVItn"
            >
              <div
                class="sc-1dzehji-0 hjLOhC"
                defaults="[object Object]"
                style="height: 100%; width: 100%;"
              >
                <span
                  class="sc-1ntgl3l-0 hGagSU"
                  style="margin: 2rem 2rem; font-size: 3rem; font-weight: 1rem;"
                  textalign="initial"
                  textcolor="inherit"
                >
                  A
                  D
                </span>
              </div>
            </li>
            <div
              bg="white"
              borderradius="2"
              class="sc-12ecrgm-0 dAEUnm sc-1298oxm-8 Hlhle"
              color="black"
              display="flex"
              font-variant="body2"
              px="4"
              py="4"
            >
              <h4
                class="sc-1ntgl3l-0 igfIhV"
                textalign="center"
                textcolor="inherit"
                textvariant="h4"
              >
                ABC
                 
                DEF
                 
                (SR0022)
              </h4>
              <h5
                class="sc-1ntgl3l-0 dBSnDZ sc-1298oxm-7 kNgBfF"
                textalign="center"
                textcolor="inherit"
                textvariant="h5"
                textweight="Medium"
              >
                Team: 
                <span>
                  BA
                </span>
              </h5>
              <h5
                class="sc-1ntgl3l-0 dBSnDZ sc-1298oxm-7 kNgBfF"
                textalign="center"
                textcolor="inherit"
                textvariant="h5"
                textweight="Medium"
              >
                Designation: 
                <span>
                  BA - Traineee
                </span>
              </h5>
              <h5
                class="sc-1ntgl3l-0 dBSnDZ sc-1298oxm-7 kNgBfF"
                textalign="center"
                textcolor="inherit"
                textvariant="h5"
                textweight="Medium"
              >
                Role: 
                <span>
                  Manager
                </span>
              </h5>
              <li
                class="sc-1298oxm-5 jElgbs"
              >
                <div
                  class="sc-1ep6kh8-0 hrIcnd"
                  id="medly-popover-wrapper"
                >
                  <h5
                    class="sc-1ntgl3l-0 dBSnDZ sc-1298oxm-7 kNgBfF"
                    textalign="center"
                    textcolor="inherit"
                    textvariant="h5"
                    textweight="Medium"
                  >
                    abc@scalereal.com
                  </h5>
                </div>
              </li>
              <h5
                class="sc-1ntgl3l-0 dBSnDZ sc-1298oxm-7 kNgBfF"
                textalign="center"
                textcolor="inherit"
                textvariant="h5"
                textweight="Medium"
              >
                +917060753645
              </h5>
            </div>
            <li
              class="sc-1298oxm-6 bKQgdL"
            >
              <button
                class="sc-1k2druj-0 dsEKPA"
                edges="square"
                type="button"
                variant="outlined"
              >
                <span
                  class="sc-1ntgl3l-0 eOsHHD"
                  textalign="initial"
                  textcolor="inherit"
                  textvariant="button1"
                >
                  Sign Out
                </span>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
</div>
`;
